Resources/public/js/views/services/plugins/ez-copycontentplugin.js

```diff
@@ -0,0 +1,150 @@
+     * a content under the selected location, on copyAction event.
+     *
+     * @namespace eZ.Plugin
+     * @class CopyContent
+     * @constructor
+     * @extends eZ.Plugin.ViewServiceBase
+     */
+    Y.eZ.Plugin.CopyContent = Y.Base.create('copyContentPlugin', Y.eZ.Plugin.ViewServiceBase, [], {
+        initializer: function () {
+            this.onHostEvent('*:copyAction', Y.bind(this._copySelectLocation, this));
+        },
+
+        /**
+         * copyAction event handler, launch the universal discovery widget
+         * to choose a location to copy the content
+         *
+         * @method _copySelectLocation
+         * @protected
+         * @param {EventFacade} e
+         */
+        _copySelectLocation: function (e) {
+            var service = this.get('host');
+
+            service.fire('contentDiscover', {
+                config: {
+                    title: "Select the location you want to copy your content into",
+                    contentDiscoveredHandler: Y.bind(this._copyContent, this),
+                },
+            });
+        },
+
+        /**
+         * Copy the content to the selected location
+         *
+         * @method _copyContent
+         * @protected
+         * @param {EventFacade} e
+         */
+        _copyContent: function (e) {
+            var service = this.get('host'),
+                app = service.get('app'),
+                parentLocationId = e.selection.location.get('id'),
+                locationId = service.get('location').get('id'),
+                that = this,
+                contentName =  service.get('content').get('name'),
+                parentContentName = e.selection.content.get('name'),
+                notificationIdentifier =  'copy-notification-' + parentLocationId + '-' + locationId;
+
+            this._notify(
+                "'" + contentName + "' is being copied under '" + parentContentName + "'",
+                notificationIdentifier,
+                'started',
+                5
+            );
+            app.set('loading', true);
+            service.get('content').copy({api: service.get('capi')}, parentLocationId, function (error, response) {
+                if (error) {
+                    that._notify('An error occured while copying your content', notificationIdentifier, 'error', 0);
+                    app.set('loading', false);
+                    return;
+                }
+                that._notify(
+                    "'" + contentName + "' has been successfully copied under '" + parentContentName + "'",
+                    notificationIdentifier,
+                    'done',
+                    5
+                );
+                that._redirect(response.getHeader('location'));
+            });
+        },
+
+        /**
+         * Redirect the user to the copied content
+         *
+         * @method _redirect
+         * @param location
+         * @protected
+         */
+        _redirect: function (location) {
```

Comment:
this method expects a location id in parameter, so let's call the parameter `locationId`